### PR TITLE
Include hero shelves in /shelves/ response

### LINKED ISSUE #16309: return hero shelves data under /shelves/
We have a new /shelves/ api endpoint but to get the primary and secondary hero shelves you have to make a separate call to /hero/.  Given the end objective to have a single API request to populate the homepage with content it makes sense to return everything at once in a single call.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #16309 @bobsilverberg does the api response seem reasonable?

### CODE REVIEW

File: src/olympia/shelves/tests/test_views.py [hunk 1]
```diff
@@ -168,6 +178,37 @@ def test_only_enabled_shelves_in_view(self):
         )
         assert result['results'][1]['addons'][0]['type'] == 'extension'
 
+    # If we delete HeroShelvesView move all the TestHeroShelvesView tests here
+    def test_hero_shelves_in_response(self):
+        phero = PrimaryHero.objects.create(
```
Comment (bobsilverberg) [inline src/olympia/shelves/tests/test_views.py:183]: I read that as a single word "phero", like "fero". Not a comment to address. Just something I found funny. Same with "shero", below (is that like a female hero)?

**Review by bobsilverberg — APPROVED** [review]
The API looks fine to me, thanks for asking! One question about the patch: should we have a test where we have data for both homepage shelves and hero shelves? It looks like the test added for heroes doesn't include any homepage shelves.

**Comment by eviljeff** [comment]
> One question about the patch: should we have a test where we have data for both homepage shelves and hero shelves? It looks like the test added for heroes doesn't include any homepage shelves.

Yeah... I thought it'd be unnecessary as both scenarios were tested but that kind of thing has bitten us in this api before so I better add one.

**Comment by bobsilverberg** [comment]
> > One question about the patch: should we have a test where we have data for both homepage shelves and hero shelves? It looks like the test added for heroes doesn't include any homepage shelves.
> 
> Yeah... I thought it'd be unnecessary as both scenarios were tested but that kind of thing has bitten us in this api before so I better add one.

Bitten us in the api. 🤣

**Pull request merged**